src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EServicesApiServiceImpl.scala

```diff
@@ -901,4 +902,144 @@ final case class EServicesApiServiceImpl(
+      eService.mode match {
+        case CatalogProcess.EServiceMode.RECEIVE if eService.riskAnalysis.isEmpty =>
+          Future.failed(EServiceRiskAnalysisIsRequired(eService.id))
+        case _                                                                    => Future.unit
+      }
+
+    def verifyExportEligibility(descriptor: CatalogProcess.EServiceDescriptor): Future[Unit] =
+      descriptor.state match {
+        case CatalogProcess.EServiceDescriptorState.DRAFT =>
+          Future.failed(NotValidDescriptor(descriptor.id.toString, descriptor.state.toString))
+        case _                                            => Future.unit
+      }
+
+    def createZip(
+      folderName: String,
+      docs: Seq[(Array[Byte], String)],
+      interfaceFile: (Array[Byte], String),
+      config: Json
+    ): Array[Byte] = {
+      val bos    = new ByteArrayOutputStream()
+      val zipOut = new ZipOutputStream(bos)
+
+      docs.foreach { case (data, entryName) =>
+        zipOut.putNextEntry(new ZipEntry(s"$folderName/documents/$entryName"))
+        zipOut.write(data)
+        zipOut.closeEntry()
+      }
+
+      val (interfaceData, interfaceFileName) = interfaceFile
+      zipOut.putNextEntry(new ZipEntry(s"$folderName/$interfaceFileName"))
+      zipOut.write(interfaceData)
+      zipOut.closeEntry()
+
+      zipOut.putNextEntry(new ZipEntry(s"$folderName/configuration.json"))
+      zipOut.write(config.toString.getBytes("UTF-8"))
+      zipOut.closeEntry()
+
+      zipOut.close()
+      bos.toByteArray
+    }
+
+    val result: Future[(String, HttpEntity.Strict)] = for {
+      organizationId <- getOrganizationIdFutureUUID(contexts)
+      eServiceUuid   <- eserviceId.toFutureUUID
+      descriptorUuid <- descriptorId.toFutureUUID
+      eService       <- catalogProcessService.getEServiceById(eServiceUuid)
+      _              <- assertRequesterAllowed(eService.producerId)(organizationId)
+      _              <- checkRiskAnalysisForExport(eService)
+      descriptor     <- eService.descriptors
+        .find(_.id === descriptorUuid)
+        .toFuture(EServiceDescriptorNotFound(eService.id.toString, descriptorId))
+      _              <- verifyExportEligibility(descriptor)
+      interface      <- descriptor.interface.toFuture(MissingInterface(eServiceUuid, descriptorUuid))
+      interfaceFile  <- fileManager
+        .get(ApplicationConfiguration.eServiceDocumentsContainer)(interface.path)
+        .map(_.toByteArray)
+      _              <- FileManagerUtils
+        .verify(interfaceFile, interface.name, eService, isInterface = true)
+        .toFuture
+      docFiles       <- descriptor.docs.traverse(doc =>
+        fileManager
+          .get(ApplicationConfiguration.eServiceDocumentsContainer)(doc.path)
+          .map(filestream => (filestream.toByteArray, doc.name))
+      )
+      config     = extractConfig(eService, descriptor)
+      folderName = s"${eService.name}_V${}${descriptor.version}"
```

Comment:
as per discussion on Slack
```suggestion
      folderName = s"${eService.id}_${descriptor.id}"
```